src/Nancy/NancyEngine.cs

```diff
@@ -187,13 +187,23 @@ private void CheckStatusCodeHandler(NancyContext context)
                 return;
             }
 
-            foreach (var statusCodeHandler in this.statusCodeHandlers)
+            var handlers = this.statusCodeHandlers
+                .Where(x => x.HandlesStatusCode(context.Response.StatusCode, context))
+                .ToList();
+
+            var defaultHandler = handlers
+                .FirstOrDefault(x => x is DefaultStatusCodeHandler);
+
+            var customHandler = handlers
+                .FirstOrDefault(x => !(x is DefaultStatusCodeHandler));
+
+            var handler = customHandler ?? defaultHandler;
+            if (handler == null)

```

Comment:
If the `DefaultStatusCodeHandler` has been removed, it could be `null`. Better safe than sorry :wink: